Assets/LeapMotion/Scripts/DetectionUtilities/Detector.cs

```diff
@@ -0,0 +1,53 @@
+﻿using UnityEngine;
+using UnityEngine.Events;
+using System.Collections;
+using Leap;
+
+namespace Leap.Unity {
+
+  public class Detector : MonoBehaviour {
+    public bool IsActive{ get{ return _isActive;} private set { _isActive = value;}}
+    private bool _isActive = false;
+    [Tooltip("Draw this detector's Gizmos, if any. (Gizmos must be on in Unity edtor, too.)")]
+    public bool ShowGizmos = true;
+    [Tooltip("Dispatched when condition is detected.")]
+    public UnityEvent OnActivate;
+    [Tooltip("Dispatched when condition is no longer detected.")]
+    public UnityEvent OnDeactivate;
+
+    /**
+    * Invoked when this detector activates.
+    * Subclasses must call this function when the detector's conditions become true.
+    */
+    public virtual void Activate(){
+      if (!IsActive) {
+        IsActive = true;
+        OnActivate.Invoke();
+      }
+      IsActive = true;
+    }
+
+    /**
+    * Invoked when this detector deactivates.
+    * Subclasses must call this function when the detector's conditions change from true to false.
+    */
+    public virtual void Deactivate(){
+      if (IsActive) {
+        IsActive = false;
+        OnDeactivate.Invoke();
+      }
+      IsActive = false;
+    }
+  }
+
+  /** 
+  * Settings for handling pointing conditions
```

Comment:
Does this belong in the base class? I don't think it would apply in all cases for extending detectors.